contracts/masset/BasketManager.sol

```diff
@@ -239,8 +240,14 @@ contract BasketManager is
         // foreach bAsset
         for(uint8 i = 0; i < count; i++) {
             Basset memory b = allBassets[i];
-            // call each integration to `checkBalance`
-            uint256 balance = IPlatformIntegration(integrations[i]).checkBalance(b.addr);
+            address bAsset = b.addr;
+
+            // call each integration to `checkBalance` and sum with the cache balance
+            address integration = integrations[i];
+            uint256 lending = IPlatformIntegration(integration).checkBalance(bAsset);
+            uint256 cache = IERC20(bAsset).balanceOf(integration);
```

Comment:
Consider changing this line to something like this to account for assets that might be transferred to the integration:
```
uint256 cache = 0;
if (!b.isTransferFeeCharged) {
  cache = IERC20(bAsset).balanceOf(integration);
}
```

Reply:
You are right. Done.